### PR TITLE
PWGHF: Add possibility to have a mass cut on correlatorDMesonPairsTesting

### PR DESCRIPTION
**Pull request by atavirag**

### CODE REVIEW

**Review by fgrosa — APPROVED** [review]
This is good for me, thanks @atavirag!

File: PWGHF/HFC/TableProducer/correlatorDMesonPairsTesting.cxx [hunk 1]
```diff
@@ -311,11 +313,16 @@ struct HfCorrelatorDMesonPairsTesting {
   /// Fill counters for D0 and D0bar
   /// \param selectedD0Candidates contains all D0 candidates
   template <typename T>
-  void GetCountersPerEvent(const T& selectedD0Candidates)
+  void GetCountersPerEvent(const T& selectedD0Candidates, const bool& selectSignalRegionOnly)
```
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorDMesonPairsTesting.cxx:316]: @atavirag For my information, why is this addition needed?

Comment (fgrosa) [reply PWGHF/HFC/TableProducer/correlatorDMesonPairsTesting.cxx:316]: Indeed there is no need to have `selectSignalRegionOnly` as argument of the function, this has slipped my eyes when looking at the code. @atavirag can you remove it in another PR? Thanks!

Comment (atavirag) [reply PWGHF/HFC/TableProducer/correlatorDMesonPairsTesting.cxx:316]: Hi, sorry, I just made the PR with the correction. It's #4732

Comment (vkucera) [reply PWGHF/HFC/TableProducer/correlatorDMesonPairsTesting.cxx:316]: Thanks
